sycl/doc/extensions/ExplicitSIMD/dpcpp-explicit-simd.md

```diff
@@ -0,0 +1,776 @@
+- ```replicate<REP, VS, W, HS>(uint16_t i=0 )```: replicate **REP** blocks of **W** sequential
+elements with a stride of **HS** starting at the i-th element from the simd vector object with
+each block strided by **VS** elements, and return a new vector of **REP** * **W** length.
+Selected blocks of **W** elements will overlap if **VS** < **W**.
+
+To avoid explicit type cast and the resulting move instructions for large vectors, Explicit SIMD allows
+programmer to reinterpret the fundamental data element type of a simd vector object and change
+its shape to 1D or 2D object through the ```format``` function:
+- ```format<EltTy>( )```: returns a reference to the calling simd object interpreted as a new
+simd vector with the size determined by the template **EltTy** parameter.
+- ```format<EltTy, Height, Width>( )```: returns a reference to the calling simd object interpreted
+as a new 2D simd_view object with the shape determined by the template parameters **Height** and**Width**. The size of the new 2D block must not exceed the size of the original object.
+
+```cpp
+  simd<int, 16> v1;
+   // ...
+  auto v2 = v1.format<short>;
+  // v2 is a reference to the location of v1
+  // interpreted as a vector of 32 shorts.
+  // ...
+  auto m1 = v1.format<int, 4, 4>;
+  // m1 is a reference to the location of v1
+  // interpreted as a matrix 4x4 of ints.
+  // ...
+  auto m2 = v1.format<char, 4, 16>( );
+  // m2 is a reference to the location of v1
+  // interpreted as a matrix 4x16 of chars.
+```
+
+To model predicated move, Explicit SIMD provides the following merge functions:
+- ```merge(value_type Val, mask_type Mask)```: this merge operation takes one source operand **Val** and a mask **Mask** defined as unsigned short vector of the same length. The semantic is that if the LSB of an element of **Mask** is set, then the corresponding data element of **Val** is copied to the corresponding position in the calling simd object.
+
+```cpp
+  simd<int, 4>   m, src;
+  simd<unsigned short, 4> mask;
+  // ...
+  m.merge(src, mask);
+  // m           src         mask      --->  m
+  // 2 2 2 2     4 4 4 4     1 1 0 1         4 4 2 4
+```
+
+- ```merge(value_type Val1, value_type Val2, mask_type Mask)```: this merge operation takes two source operands **Val1** and **Val2** as well as a simd mask. The semantic is that if the LSB of an element of **Mask** is set, then the corresponding data element of **Val1** is copied to the corresponding position in the calling simd object. Otherwise the corresponding data element of **Val2** is copied to the corresponding position in the calling simd object.
+
+```cpp
+  simd<int,4>   m, src1, src2;
+  simd<unsigned short, 4> mask;
+  // ...
+  m.merge(src1, src2, mask);
+  // m           src1        src2         mask        --->  m
+  // 2 2 2 2     4 4 4 4     3 3 3 3      1 1 0 1           4 4 3 4
+```
+### `simd_view` class
+
+The ```sycl::intel::gpu::simd_view``` represents a "window" into existing simd object,
```

Comment:
How is it different from your `select` above?
I prefer this `simd_view` naming.